lib/test_formulae.rb

```diff
@@ -132,38 +134,43 @@ def download_artifact_from_previous_run!(artifact_name, dry_run:)
         check_suite_nodes = response.dig("repository", "object", "checkSuites", "nodes")
         return if check_suite_nodes.blank?
 
-        wanted_artifact = artifact_metadata(check_suite_nodes, github_repository, "pull_request",
-                                            "CI", "conclusion", artifact_name)
-        # If we didn't find the artifact that we wanted, fall back to the `event_payload` artifact.
-        wanted_artifact ||= artifact_metadata(check_suite_nodes, github_repository, "pull_request_target",
-                                              "Triage tasks", "upload-metadata", "event_payload")
-        return if wanted_artifact.blank?
+        wanted_artifacts = artifact_metadata(check_suite_nodes, github_repository, "pull_request",
+                                             "CI", "conclusion", artifact_pattern)
+        wanted_artifacts_pattern = artifact_pattern
+        if wanted_artifacts.empty?
+          # If we didn't find the artifacts that we wanted, fall back to the `event_payload` artifact.
+          wanted_artifacts = artifact_metadata(check_suite_nodes, github_repository, "pull_request_target",
+                                               "Triage tasks", "upload-metadata", "event_payload")
+          wanted_artifacts_pattern = "event_payload"
+        end
+        return if wanted_artifacts.empty?
 
-        wanted_artifact_name = wanted_artifact.fetch("name")
-        if @downloaded_artifacts[sha].include?(wanted_artifact_name)
-          opoo "Already tried #{wanted_artifact_name} from #{sha}, giving up"
+        if @downloaded_artifacts[sha].include?(wanted_artifacts_pattern)
+          opoo "Already tried #{wanted_artifacts_pattern} from #{sha}, giving up"
           return
         end
```

Comment:
Should we have similar handling for specific artifact names, and not just patterns? (Or only this handling for specific names and not patterns?)